static/js/containers/App.js

```diff
@@ -57,6 +87,48 @@ class App extends React.Component {
     }
   }
 
+  requireCompleteProfile() {
+    const {
+      userProfile,
+      userProfile: { profile },
+      location: { pathname },
+      dispatch,
+    } = this.props;
+    let profileCompleteCheck = profileComplete(profile);
+    if (
+      userProfile.getStatus === FETCH_SUCCESS &&
+      profile.agreed_to_terms_of_service &&
+      !PROFILE_REGEX.test(pathname) &&
+      profileCompleteCheck !== true
+    ) {
+      const { url, title, text } = profileCompleteCheck;
+      dispatch(updateDialogText(text));
+      dispatch(updateDialogTitle(title));
+      dispatch(setDialogVisibility(true));
+      this.context.router.push(url);
+    }
+  }
+
+  showDialog () {
```

Comment:
We should always show the dialog and use the `open` prop of `Dialog` to say where it is open or not. Also, we will need to use `dialogPolyfill` here to support Firefox and IE